tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterBuilderTests.cs

```diff
@@ -36,20 +37,21 @@ public class ClusterBuilderTests
         public void CreateServerMonitorFactory_should_return_expected_result(int connectTimeoutMilliseconds, int heartbeatTimeoutMilliseconds, int expectedServerMonitorConnectTimeoutMilliseconds, int expectedServerMonitorSocketTimeoutMilliseconds)
         {
             var connectTimeout = TimeSpan.FromMilliseconds(connectTimeoutMilliseconds);
-            var authenticators = new[] { new DefaultAuthenticator(new UsernamePasswordCredential("source", "username", "password")) };
+            var authenticatorFactory = new AuthenticatorFactory(() => new DefaultAuthenticator(new UsernamePasswordCredential("source", "username", "password")));
             var heartbeatTimeout = TimeSpan.FromMilliseconds(heartbeatTimeoutMilliseconds);
             var expectedServerMonitorConnectTimeout = TimeSpan.FromMilliseconds(expectedServerMonitorConnectTimeoutMilliseconds);
             var expectedServerMonitorSocketTimeout = TimeSpan.FromMilliseconds(expectedServerMonitorSocketTimeoutMilliseconds);
             var subject = new ClusterBuilder()
                 .ConfigureTcp(s => s.With(connectTimeout: connectTimeout))
-                .ConfigureConnection(s => s.With(authenticators: authenticators))
+                .ConfigureConnection(s => s.With(authenticatorFactories: new[] { authenticatorFactory }))
                 .ConfigureServer(s => s.With(heartbeatTimeout: heartbeatTimeout));
 
             var result = (ServerMonitorFactory)subject.CreateServerMonitorFactory();
 
             var serverMonitorConnectionFactory = (BinaryConnectionFactory)result._connectionFactory();
             var serverMonitorConnectionSettings = serverMonitorConnectionFactory._settings();
-            serverMonitorConnectionSettings.Authenticators.Should().HaveCount(0);
+            var authenticators = serverMonitorConnectionSettings.AuthenticatorFactories.Select(a => a.Create());
+            authenticators.Should().HaveCount(0);
```

Comment:
Replace 53-54 with:

            serverMonitorConnectionSettings.AuthenticatorFactories.Should().HaveCount(0);